kuksa-persistence-provider/src/kuksaconnector.rs

```diff
@@ -0,0 +1,458 @@
+* Copyright (c) 2024 Contributors to the Eclipse Foundation
+*
+* This program and the accompanying materials are made available under the
+* terms of the Apache License 2.0 which is available at
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* SPDX-License-Identifier: Apache-2.0
+********************************************************************************/
+
+use crate::storage::{self, StoreItem};
+
+use std::collections::HashMap;
+use std::fmt;
+use std::time::SystemTime;
+
+use kuksa::proto;
+
+use std::sync::mpsc::Sender;
+
+use std::sync::{Arc, Mutex};
+
+#[derive(Debug)]
+pub struct ParseError {}
+
+pub fn create_kuksa_client(uri: &str) -> Arc<Mutex<kuksa::Client>> {
+    log::info!("Creating Kuksa Databroker client for URI: {}", uri);
+    let uri = kuksa::Uri::try_from(uri).expect("Invalid URI  for Kuksa Databroker connection.");
+    Arc::new(Mutex::new(kuksa::Client::new(uri)))
+}
+
+pub async fn get_from_storage_and_set_values(
+    storage: &impl storage::Storage,
+    kuksa_client: &Arc<Mutex<kuksa::Client>>,
+    vsspaths: &Vec<String>,
+) {
+    for vsspath in vsspaths {
+        get_from_storage_and_set(storage, kuksa_client, vsspath, false).await;
+    }
+}
+
+pub async fn get_from_storage_and_set_actuations(
+    storage: &impl storage::Storage,
+    kuksa_client: &Arc<Mutex<kuksa::Client>>,
+    vsspaths: &Vec<String>,
+) {
+    for vsspath in vsspaths {
+        get_from_storage_and_set(storage, kuksa_client, vsspath, true).await;
+    }
+}
+
+pub async fn get_from_storage_and_set(
+    storage: &impl storage::Storage,
+    kuksa_client: &Arc<Mutex<kuksa::Client>>,
+    vsspath: &str,
+    is_actuator: bool,
+) {
+    log::debug!("Query storage for VSS signal: {}", vsspath);
+    let value = match storage.get(vsspath) {
+        Some(x) => x,
+        None => {
+            log::warn!("No value for VSS signal: {} stored", vsspath);
+            return;
+        }
+    };
+
+    //Figure out metadata:
+    let datapoint_entries = match kuksa_client
+        .lock()
+        .unwrap()
+        .get_metadata(vec![vsspath])
+        .await
+    {
+        Ok(data_entries) => Some(data_entries),
+        Err(kuksa::Error::Status(status)) => {
+            log::warn!(
+                "Error: Could not get metadata for VSS signal: {}, Status: {}",
+                vsspath,
+                &status
+            );
+            None
+        }
+        Err(kuksa::Error::Connection(msg)) => {
+            log::warn!(
+                "Connection Error: Could not get metadata for VSS signal: {}, Reason: {}",
+                vsspath,
+                &msg
+            );
+            None
+        }
+        Err(kuksa::Error::Function(msg)) => {
+            log::warn!(
+                "Error: Could not get metadata for VSS signal: {}, Errors: {msg:?}",
+                vsspath
+            );
+            None
+        }
+    };
+
+    if datapoint_entries.is_none() {
+        return;
```

Comment:
log message?

Reply:
Is this only for error situations, i.e. we have already got one of the log messages above? Replace `None` with `return None`above?